### DISCUSSION TITLE
is it possible using the out of the box plugins to ground to a specific data source?

### DISCUSSION THREAD

**Discussion by dluc** [microsoft/semantic-kernel#6639] [Q&A] [answered]
Question from `jfarrell` on Discord:

> Hello, I just have a brief question, as I am unable to locate the information in the docs - is it possible, using the out of the box plugins (Azure OpenAI specifically) to ground the call using a specific data source?
>
> Clarification: I want to be able to use my own data, located in Azure AI Search, as the basis for any prompts against my OpenAI model referenced in my kernel.
>
>Thanks
>Presently, I am anticipating this would need to be done through a plugin which codifies my specific requirement against an Azure OpenAI endpoint,.

src: https://discord.com/channels/1063152441819942922/1086057678238928967/1249546786289483878

**Comment by sophialagerkranspandey** [comment, marked as answer]
We'd recommend using OpenAI with your data to solve your problem and integrate with AzureAISearch out of the box. Here's a sample as well: [semantic-kernel/dotnet/samples/Concepts/ChatCompletion/AzureOpenAIWithData_ChatCompletion.cs at main · microsoft/semantic-kernel (github.com)](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/Concepts/ChatCompletion/AzureOpenAIWithData_ChatCompletion.cs)
